tests/packages/index.js

```diff
@@ -0,0 +1,23 @@
+let glob = require("glob").sync
+let cp = require('child_process')
+let path = require('path');
+
+function exec(commands) {
+    glob(__dirname+"/**/packages/*").forEach((_path) => {
+        let buildcommands = require(path.join(_path, 'package.json')).scripts.build.split(" && ")
+        for (let cmds of buildcommands){
+            let child = cp.spawn('node', cmds.split(" "), { env: process.env, cwd: _path, stdio: 'inherit' })
+                .on("error", (error) => {console.error(error); process.exit(1)})
+                .on("close", (code, signal) => {
+                if (code != 0 ){
+                    console.error(path.basename(_path) + " Failed");
+                    process.exit(1);
+                }else {
+                    console.log(path.basename(_path) + " Passed");
+                }});
+
+        }
+    });
+}
+
+exec(process.argv.slice(2))
```

Comment:
Still not quite happy with this mechanism. Would prefer a more declarative approach instead of specialized code. Like, wdyt of another `package.json` in `tests/packages` instead, with all the necessary scripts, that is then run from the main package.json via something like `"test:packages": "cd tests/packages && npm run all && cd ../.."`?

Reply:
Well I had avoided that because I didn't think that `cd` worked on windows and why I made the scripts in the first place.  I'll try that now and you'll have to let me know if it works.

Reply:
I had avoided doing this as I thought it wouldn't work on windows.  I changed back to it; let me know if it doesn't.

Reply:
Yeah, `cd` is as universal as it can get, except when options are required.